src/pci_lmt/results.py

```diff
@@ -84,3 +102,33 @@ def write(self, result: LmtLaneResult) -> None:
             self.__emitted_header = True
 
         print(row)
+
+
+class OctTvOutputReporter(Reporter):
+    def __init__(self):
+        # TODO(sksekar): Add support for saving to file.
+        self._run: tv.TestRun = None
+        self._step: tv.TestStep = None
```

Comment:
a single variable here makes the assumption that you have a single step. consider the following application

```
r = OcptvReporter()
r.start_run()  # start run0
r.start_step()  # start step0
r.start_step()  # start step1
r.end_step()  # end step1, with NO WAY of ending step 0, which is lost; steps, as well, cannot be nested
r.end_run()  # end run0
```

the way this is avoided in ocptv api is that only the `run` objects can create steps and the steps are scoped (as implied by their reference binding)

Reply:
Good catch. Looks like one way to avoid this is to implement the start_run/start_step as contextmanagers and invoke end_run/end_step as part of automatic cleanup. Would like to get your inputs before diving deeper.

Reply:
yes, this is an option i highlighted above. examples https://github.com/opencomputeproject/ocp-diag-core-python/blob/dev/examples/measurements.py

Reply:
Thanks for the suggestion and pointers. I've incorporated your comments.